tests/core/calculate/test_ase_calculator.py

```diff
@@ -0,0 +1,305 @@
+            if "omol" in predict_unit.datasets:
+                yield FAIRChemCalculator(predict_unit, task_name="omol")
+
+    return _calc_generator
+
+
+@pytest.fixture()
+def slab_atoms() -> Atoms:
+    atoms = fcc111("Pt", size=(2, 2, 5), vacuum=10.0, periodic=True)
+    add_adsorbate(atoms, "O", height=1.2, position="fcc")
+    atoms.pbc = True
+    return atoms
+
+
+@pytest.fixture()
+def bulk_atoms() -> Atoms:
+    return bulk("Fe", "bcc", a=2.87).repeat((2, 2, 2))
+
+
+@pytest.fixture()
+def aperiodic_atoms() -> Atoms:
+    return molecule("H2O")
+
+
+@pytest.fixture()
+def periodic_h2o_atoms() -> Atoms:
+    """Create a periodic box of H2O molecules."""
+    atoms = molecule("H2O")
+    atoms.set_cell([100.0, 100.0, 100.0])  # Define a cubic cell
+    atoms.set_pbc(True)  # Enable periodic boundary conditions
+    atoms = atoms.repeat((2, 2, 2))  # Create a 2x2x2 periodic box
+    return atoms
+
+
+@pytest.fixture()
+def large_bulk_atoms() -> Atoms:
+    """Create a bulk system with approximately 1000 atoms."""
+    return bulk("Fe", "bcc", a=2.87).repeat((10, 10, 10))  # 10x10x10 unit cell
+
+
+@pytest.mark.gpu()
+def test_calculator_setup(all_calculators):
+    for calc in all_calculators():
+        implemented_properties = ["energy", "forces"]
+
+        # all conservative UMA checkpoints should support E/F/S!
+        if (
+            not calc.predictor.direct_forces
+            and len(calc.predictor.datasets) > 1
+            or calc.task_name != "omol"
+        ):
+            print(len(calc.predictor.datasets), calc.task_name)
+            implemented_properties.append("stress")
+
+        assert all(
+            prop in calc.implemented_properties for prop in implemented_properties
+        )
+        assert all(
+            prop in calc.calc_property_to_model_key_mapping
+            for prop in implemented_properties
+        )
+
+
+@pytest.mark.gpu()
+@pytest.mark.parametrize(
+    "atoms_fixture",
+    [
+        "slab_atoms",
+        "bulk_atoms",
+        "aperiodic_atoms",
+        "periodic_h2o_atoms",
+        "large_bulk_atoms",
+    ],
+)
+def test_energy_calculation(request, atoms_fixture, all_calculators):
+    for calc in all_calculators():
+        atoms = request.getfixturevalue(atoms_fixture)
+        atoms.calc = calc
+        energy = atoms.get_potential_energy()
+        assert isinstance(energy, float)
+
+
+@pytest.mark.gpu()
+def test_relaxation_final_energy(slab_atoms, mlip_predict_unit):
+    calc = FAIRChemCalculator(
+        mlip_predict_unit,
+        task_name=mlip_predict_unit.datasets[0],
+    )
+
+    slab_atoms.calc = calc
+    initial_energy = slab_atoms.get_potential_energy()
+    assert isinstance(initial_energy, float)
+
+    opt = BFGS(slab_atoms)
+    opt.run(fmax=0.05, steps=100)
+    final_energy = slab_atoms.get_potential_energy()
+    assert isinstance(final_energy, float)
+    assert final_energy <= initial_energy
```

Comment:
```suggestion
```

Since the task being selected not necessarily do well for slabs, it's likely the relaxation is nonsense and this is not true. Especially for omol models.